tests/test_functional.py

```diff
@@ -87,6 +88,92 @@ def test_overlay(self, dir):
         compare(config.user, expected=2)
         compare(config.file, expected=3)
 
+    def test_lazy_load(self, dir):
+        yaml = pytest.importorskip("yaml")
+
+        class VaultClient:
+
+            def __init__(self, host, token):
+                self.host = host
+                self.token = token
+
+            def load(self, key):
+                return {'name': key.capitalize(), 'password': '...'}
+
+        class VaultValue(DataValue):
+
+            def __init__(self, proxy, key):
+                self.proxy = proxy
+                self.key = key
+
+            def get(self):
+                if self.proxy.client is None:
+                    raise RuntimeError('Cannot use proxy before a client is configured')
+                return self.proxy.client.load(self.key)
+
+        class VaultProxy:
+
+            client = None
+
+            def parser(self):
+
+                class Loader(yaml.Loader):
+                    pass
+
+                Loader.add_constructor('!from_vault', self.value_from_yaml)
+
+                def parser(path):
+                    return yaml.load(path, Loader)
+
+                return parser
+
+            def value_from_yaml(self, loader, node):
+                print(self)
+                return VaultValue(self, loader.construct_mapping(node)['key'])
+
+            def __deepcopy__(self, memodict):
+                # This is required as an artifact of how merging/cloning work...
+                # It's ugly, but what would work better?
+                return self
```

Comment:
This is a key reason why I wanted to expose the path and the root config instead of making it magically integrate it into the loader

Reply:
Ah, now I understand what the paths in from_vault are! Sorry, missed that before.
I'm not sure it helps, since setting `.vault_loader` on the root config still ends up with it in `.data` and so the clone problem will reappear later.

Also, its seems superfluous to have to specify the config path where you want the password to be put, given that your `!from_vault` is already in exactly that location. What happens if you get the path wrong?

I'm going to play with having `Proxy` be a configurator thing so `__deepcopy__` becomes an implementation detail.